#!/usr/bin/env python
PACKAGE = 'chatbot'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("enable", bool_t, 0, "Enable Chatbot", True)
gen.add("mute", bool_t, 0, "Mute the Chatbot, chatbot still responds but won't send result to tts", False)
gen.add("insert_behavior", bool_t, 0, "Automatically insert non-verbal behaviors", False)
gen.add("chatbot_url", str_t, 0, "Chatbot Server URL", 'http://localhost:8001')
gen.add("sentiment", bool_t, 0, "Enable Sentiment", True)
gen.add("delay_response", bool_t, 0, "Delay Chatbot Response", False)
gen.add("delay_time", double_t, 0, "Response Delay Time", 5, 0, 10)
gen.add("ignore_indicator", bool_t, 0, "Ignore the indicator from response", False)
gen.add("set_that", str_t, 0, "Set 'that' tag", '')
gen.add("set_context", str_t, 0, "Add Chatbot Context (k=v,k2=v2,...)", '')

#marker_enum = gen.enum([
#        gen.const("Default", str_t, "default", ""),
#        gen.const("Demo", str_t, "demo", ""),
#        gen.const("Test", str_t, "test", ""),
#    ], "Markers")
#gen.add("marker", str_t, 0, "Marker", "default", edit_method=marker_enum)
gen.add("marker", str_t, 0, "Set marker, demo name, test or any other string that marks this dialogue", 'default')

# package name, node name, config name
exit(gen.generate(PACKAGE, "chatbot", "Chatbot"))
